dgp/gui/plotting/mplutils.py

```diff
@@ -380,6 +390,95 @@ def reset_view(self, x_margin=None, y_margin=None):
         self.set_xlim(min_x0, max_x1)
 
 
+class PatchManager:
+    def __init__(self, parent=None):
+        self.patchgroups = []  # type: List[RectanglePatchGroup]
+        self._active = None
+        self._x0 = None  # X location when active group was selected
+        self.parent = parent
+
+    @property
+    def active(self) -> Union[None, 'RectanglePatchGroup']:
+        return self._active
+
+    @property
+    def groups(self):
+        """Return a sorted list of patchgroups by patch x location."""
+        return sorted(self.patchgroups, key=lambda pg: pg.x)
+
+    def valid_click(self, xdata, proximity=0.05):
+        """Return True if xdata is a valid location to place a new patch
+        group, False if it is too close to an existing patch."""
+        pass
+
+    def add_group(self, group: 'RectanglePatchGroup'):
+        self.patchgroups.append(group)
+
+    def select(self, xdata, inner=True) -> bool:
+        self.deselect()
+        for pg in self.groups:
+            if xdata in pg:
+                pg.animate(xdata)
+                self._active = pg
+                self._x0 = xdata
+                break
+        else:
+            self._x0 = None
+
+        return self._active is not None
+
+    def deselect(self) -> None:
+        if self._active is not None:
+            self._active.unanimate()
+        self._active = None
+
+    def rescale_patches(self):
+        for group in self.patchgroups:
+            group.fit_height()
+
+    def onmotion(self, event: MouseEvent) -> None:
+        if event.xdata is None:
+            return
+        if self.active is None:
+            self.highlight_edge(event.xdata)
+            event.canvas.draw()
+        else:
+            dx = event.xdata - self._x0
+            self.active.shift_x(dx)
+
+    def highlight_edge(self, xdata: float) -> None:
+        """
+        Called on motion event if a patch isn't selected. Highlight the edge
+        of a patch if it is under the mouse location.
+        Return all other edges to black
+
+        Parameters
+        ----------
+        xdata : float
+            Mouse x-location in plot data coordinates
+
+        """
+        edge_grp = None
+        self.parent.setCursor(QtCore.Qt.ArrowCursor)
+        for group in self.groups:
+            edge = group.get_edge(xdata, inner=False)
+            if edge in ('left', 'right'):
+
+                edge_grp = group
+                self.parent.setCursor(QtCore.Qt.SizeHorCursor)
+                group.set_edge(edge, 'red', select=False)
+                break
+        else:
+            # group.set_edge('', 'black', select=False)
```

Comment:
Still need this commented code?

Reply:
Nope, cleaned that out. In fact I'm not sure about the future of the mplutils classes, they might be useful for plotting transformed data, but like below with plotters.py will have to consider whether to keep them around.